stone/backends/python_type_stubs.py

```diff
@@ -334,7 +334,7 @@ def _generate_struct_class_init(self, ns, struct):
             field_name_reserved_check = fmt_var(field.name, True)
             field_type = self.map_stone_type_to_pep484_type(ns, field.data_type)
 
-            if not is_nullable_type(field.data_type):
+            if field.has_default:
```

Comment:
Do Nullable types have `has_default` as true? 

Why not just make it `if is_nullable_type(field.data_type)`

Reply:
Yes Nullable types have default values https://github.com/dropbox/stone/blob/7aad3c33cb890bd728ddd21976c867262663d943/stone/backends/python_rsrc/stone_validators.py#L618-L622

There is another case beside Nullable types. Types with default values that are not `None` should also be optional. `field.has_default` captures both cases